modules/openapi-generator/src/main/resources/kotlin-client/libraries/jvm/infrastructure/ApiClient.kt.mustache

```diff
@@ -90,7 +90,7 @@ import java.io.File
         }
 
     protected inline fun <reified T: Any?> responseBody(body: ResponseBody?, mediaType: String? = JsonMediaType): T? {
-        if(body == null) {
+        if(body == null || mediaType == null) {
```

Comment:
This is potentially "breaking". Rather than throwing an unsupported error for invalid media type, it'll now result in null without providing further details to the user.

If the intent is to return null for unsupported media types, can this be moved to another if block and add a warning log message?

Reply:
Same for all ApiClient mustache changes.

Reply:
Sure I can do that. Would you reckon an error log like "mediaType is null" is sufficient or would you prefer some more details in the log?

Reply:
I don't think "mediaType is null" would work here, because this isn't a value which would be null because of programmatic error.

This argument is pulled from the 'Content-Type' header. If one doesn't exist for whatever reason, this would just be null and that could be confusing.

I think something explaining that media type is null and we expected Json at this point?

Again, I'm not following exactly what the intent is with the early return if media type is null. That's covered by the last block of the method. If we're going to "swallow" a response, we'll want to explain why. For body, it makes sense because a null body means a null response. I hope that helps clarify my concern.

Reply:
I actually had cases where a certain api provided a null media type but the body was not null. But I agree that these cases would rather confuse than help. After thinking about that `mediaType == null`-condition, I think the best would be to leave it out (also to avoid breaking changes). I will do some more research and will provide a new PR (if needed).